CoreScriptsRoot/Modules/DeveloperConsoleModule.lua

```diff
@@ -3015,4 +3024,46 @@ function DevConsoleModuleTable:SetVisibility(value)
 	devConsole:SetVisible(value)
 end
 
+
+local creatingLock = false
+local creatingVisibleValueToSet = false
+
+local function SetCoreConsoleCreation()
+	if (creatingLock) then return end
+	creatingLock = true
+
+	spawn(function()
+		--// Keep GetVisibility call before SetVisibility because the first call will yield for some time and 
+		--// there is the possibility that during the yield time the value of 'creatingVisibleValueToSet' may
+		--// change.
+		DevConsoleModuleTable:GetVisibility()
+		DevConsoleModuleTable:SetVisibility(creatingVisibleValueToSet)
+
+		creatingLock = false
+	end)
+end
+
+local StarterGui = game:GetService("StarterGui")
+StarterGui:RegisterGetCore("DeveloperConsoleVisible", function()
+	if (not myDeveloperConsole) then
+		SetCoreConsoleCreation()
+		return creatingVisibleValueToSet;
+	else
+		return DevConsoleModuleTable:GetVisibility()
+	end
+end)
+StarterGui:RegisterSetCore("DeveloperConsoleVisible", function(visible)
+	if (type(visible) ~= "boolean") then
+		warn("DeveloperConsoleVisible must be given a boolean value.")
```

Comment:
this won't give you a call stack thought right? If it is standard practice for function calls that we define (whether it is a rbx::reflection or something defined in lua) to throw an error with invalid args, its best to be consistent. Warnings are nice, but they are just warnings. If you are fundamentally using an API wrong, an error seems appropriate.